pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/MedicalRecordUtil.java

```diff
@@ -10,58 +10,81 @@
-    private static final String SQL_SELECT_MEDICAL_RECORD_ID_AND_TYPE_FOR_PARTICIPANT = "SELECT rec.medical_record_id, inst.type "
-            + "FROM ddp_institution inst, ddp_participant part, ddp_medical_record rec "
-            + "WHERE part.participant_id = inst.participant_id AND rec.institution_id = inst.institution_id AND NOT rec.deleted <=> 1 "
-            + "AND part.participant_id = ? AND inst.type = ?";
+    private static final String SQL_SELECT_MEDICAL_RECORD_ID_AND_TYPE_FOR_PARTICIPANT =
+            "SELECT rec.medical_record_id, inst.type " + "FROM ddp_institution inst, ddp_participant part, ddp_medical_record rec "
+                    + "WHERE part.participant_id = inst.participant_id AND rec.institution_id = inst.institution_id AND NOT rec.deleted <=> 1 "
+                    + "AND part.participant_id = ? AND inst.type = ?";
 
-    public static void writeNewMedicalRecordIntoDb(Connection conn, String query, String id) {
+    private static final Logger logger = LoggerFactory.getLogger(MedicalRecordUtil.class);
+
+    public static void writeNewMedicalRecordIntoDb(Connection conn, String query, String institutionId,
+                                                   String ddpParticipantId, String instanceName) {
+        Integer mrId = null;
         if (conn != null) {
-            try (PreparedStatement insertNewRecord = conn.prepareStatement(query)) {
-                insertNewRecord.setString(1, id);
+            try (PreparedStatement insertNewRecord = conn.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+                insertNewRecord.setString(1, institutionId);
                 insertNewRecord.setLong(2, System.currentTimeMillis());
                 insertNewRecord.setString(3, SystemUtil.SYSTEM);
                 int result = insertNewRecord.executeUpdate();
                 if (result > 1) { // 0 or 1 is good
                     throw new RuntimeException("Error updating row");
                 }
-                logger.info("Added new medical record for institution w/ id " + id);
+                if (result == 1) {
+                    try (ResultSet rs = insertNewRecord.getGeneratedKeys()) {
+                        if (rs.next()) { //no next if no generated return key -> update of institution timestamp does not return new key
+                            mrId = rs.getInt(1);
+                        }
+                    } catch (Exception e) {
+                        throw new RuntimeException("Error getting id of new medical record ", e);
+                    }
+                }
+                logger.info("Added new medical record for institution w/ id " + institutionId);
             } catch (SQLException e) {
                 throw new RuntimeException("Error inserting new medical record ", e);
             }
         } else {
             throw new RuntimeException("DB connection was null");
         }
+
+        if (mrId != null) {
```

Comment:
adding upsert to ES here to add the mr object to ES even if it is empty except of the medicalRecordId

Reply:
@gmakharat @Anzori-Ghurtchumelia